lib/postgres/postgres-pr/connection.rb

```diff
@@ -129,12 +129,101 @@ def peerport
     @conn.peerport
   end
 
+  def peerinfo
+    "#{peerhost}:#{peerport}"
+  end
+
   def current_database
     @params['database']
   end
 
-  def peerinfo
-    "#{peerhost}:#{peerport}"
+  # List of supported PostgreSQL platforms & architectures:
+  # https://postgrespro.com/docs/postgresql/16/supported-platforms
+  def map_compile_os_to_platform(compile_os)
+    return Msf::Platform::Unknown.realname if compile_os.blank?
+
+    compile_os = compile_os.downcase.encode(::Encoding::BINARY)
+
+    if compile_os.match?('linux')
+      platform = Msf::Platform::Linux
+    elsif compile_os.match?(/(darwin|mac|osx)/)
+      platform = Msf::Platform::OSX
+    elsif compile_os.match?('win')
+      platform = Msf::Platform::Windows
+    elsif compile_os.match?('free')
+      platform = Msf::Platform::FreeBSD
+    elsif compile_os.match?('net')
+      platform = Msf::Platform::NetBSD
+    elsif compile_os.match?('open')
+      platform = Msf::Platform::OpenBSD
+    elsif compile_os.match?('solaris')
+      platform = Msf::Platform::Solaris
+    elsif compile_os.match?('aix')
+      platform = Msf::Platform::AIX
+    elsif compile_os.match?('hpux')
+      platform = Msf::Platform::HPUX
+    elsif compile_os.match?('irix')
+      platform = Msf::Platform::Irix
+    else
+      # Return the query result if the value can't be mapped
+      return compile_os
+    end
+
+    platform.realname
+  end
+
+  # List of supported PostgreSQL platforms & architectures:
+  # https://postgrespro.com/docs/postgresql/16/supported-platforms
+  def map_compile_arch_to_architecture(compile_arch)
```

Comment:
I have bucket these by what I have covered versus not. Wanted to call this out in case anyone has thoughts or opinions.

Historical versions. Not covered and not present in `rex/arch.rb`: 
- Alpha 
- Itanium
- M32R
- M68K
- M88K 
- NS32K
- SuperH
- VAX

17.6 supported but not covered in `rex/arch.rb`:
- S/390
- RISC-V
- PA-RISC

17.6 supported and covered in `rex/arch.rb`:
- x86 
- SPARC
- ARM
- MIPS
- PowerPC